### DISCUSSION TITLE
Honor reduced motion preference

### DISCUSSION THREAD

**Discussion by dstengele** [home-assistant/frontend#22518] [Accessibility]
I am not sure if this has been a recent change, but in the current 2024.10 release, there are (in my opinion too long) animations when opening and closing modal dialogs, for example the entity view.
As modern operating systems have a function to set a preference for reduced motion in user interfaces which can also be queries in CSS (https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion), I would love for Home Assistant to honor this preference and disable most animations if it is set.
Aside from better accesibility for people with motion sickness this could also help with performance, see for example #13816 or #17941.

**Comment by ndrwrbgs** [comment]
[Setup base animation styles, add fade out to launch screen #27829](https://github.com/home-assistant/frontend/pull/27829)
[Disable energy distribution animation if prefers-reduced-motion is set #24581](https://github.com/home-assistant/frontend/pull/24581)
(Linking discussion & code trails)

**Reply by ndrwrbgs** [reply]
I was working on this a bit but hesitated as I wasn't sure there was an appetite for it.

Dropping some notes here. I think the biggest risk is that animations are configured throughout the code base right now, so making any change that would _persist_ the future, would require buy in and commitment from the core frontend team so that more don't "creep in" later.

<details>
<summary>Some of the major animation values I found throughout the codebase included...</summary>

```
most common: 180ms

15ms gallery
80ms control-select
100ms color-picker
150ms automation-row
200ms
	two-pane-top-app-bar-fixed
	ha-wa-dialog
	ha-dialog-show-duration
	ha-dialog-hide-duration
280ms
300: ha-bottom-sheets BOTTOM_SHEET_ANIMATION_DURATION_MS
350ms [default, honors reduced motion]
```
</details>
